lib/versions.nix

```
@@ -61,4 +61,17 @@ rec {
     versionSuffix = lib.removePrefix numericVersion version;
   in lib.concatStringsSep "." (lib.take n (lib.splitVersion numericVersion ++ lib.genList (_: "0") n)) + versionSuffix;
 
+  /* Format date to YYYY-MM-DD. This is usually used with a flake's input lastModifiedDate value.
+
+     Type: dateFormat :: String -> String
+
+     Example:
+       dateFormat "19700101"
+       => "1970-01-01"
+       dateFormat self.lastModifiedDate # (20230426163528)
+       => "2023-04-26"
+  */
+  dateFormat = d:
+    assert lib.assertMsg ((builtins.stringLength d) >= 8) "versions.dateFormat: date string length must be at least 8 characters long!";
+    builtins.concatStringsSep "-" (builtins.match "(.{4})(.{2})(.{2}).*" d);
```

The `.*` part is untested.

---

I'd prefer a stricter regex, so that garbage in is detected early and the feedback to the user is better.

```suggestion
    builtins.concatStringsSep "-" (builtins.match "([0-9]{4})([0-9]{2})([0-9]{2})[0-9]*" d);
```

This isn't guaranteed to match, so when `match` returns `null`, it should throw an exception that the input wasn't a `YYYYMMDD[0-9]*` date string.